docs/src/main/asciidoc/hibernate-orm-panache.adoc

```diff
@@ -330,107 +756,124 @@ params.put("status", Status.Alive);
+====
 
 == Transactions
 
-Make sure to wrap methods modifying your database (e.g. `entity.persist()`) within a transaction. Marking a
-CDI bean method `@Transactional` will do that for you and make that method a transaction boundary. We recommend doing
-so at your application entry point boundaries like your REST endpoint controllers.
+Wrap methods that modify your database, for example `entity.persist()`, within a transaction. When you do this, the JPA will batch any changes that you make to you entities at the end of the transaction or before a query. This process is called a flush. It is the most efficient way to persist changes.
 
-JPA batches changes you make to your entities and sends changes (it's called flush) at the end of the transaction or before a query.
-This is usually a good thing as it's more efficient.
-But if you want to check optimistic locking failures, do object validation right away or generally want to get immediate feedback, you can force the flush operation by calling `entity.flush()` or even use `entity.persistAndFlush()` to make it a single method call. This will allow you to catch any `PersistenceException` that could occur when JPA send those changes to the database.
-Remember, this is less efficient so don't abuse it.
-And your transaction still has to be committed.
 
-Here is an example of the usage of the flush method to allow making a specific action in case of `PersistenceException`:
+NOTE: If you want to check optimistic locking failures, perform object validation right away, or get immediate feedback, you can force the flush operation by calling `entity.flush()` or use `entity.persistAndFlush()` to make it a single method call. This enables you to catch any `PersistenceException` errors that occur when JPA sends changes to the database. However, this method is not efficient and transactions must still be committed.
+
+
+.Procedure
+
+To create a transaction, add the `@Transactional` annotation to your method at application entry point boundaries such as REST endpoint controllers.
+
+The following example demonstrates the use of the flush method to allow a specific action in case of a `PersistenceException` error:
 [source,java]
 ----
 @Transactional
 public void create(Parameter parameter){
     try {
-        //Here I use the persistAndFlush() shorthand method on a Panache repository to persist to database then flush the changes.
+        //The following code uses the persistAndFlush() shorthand method on a Panache repository to persist to the database then flush the changes:
         return parameterRepository.persistAndFlush(parameter);
     }
     catch(PersistenceException pe){
         LOG.error("Unable to create the parameter", pe);
-        //in case of error, I save it to disk
+        //The following code saves the data to disk in case of an error:
```

Comment:
I don't think we use `The following code` in code comments generally, so this feels unnatural. Is it standard procedure in code comments from docs?

Reply:
Well, in general we try not to include comments in code. So this was a compromise. However, I have now moved the comments to the introductory text.

Reply:
Interesting, because devs are encouraged and used to comments in the code to explain what's going on.